contracts/Vault1.sol

```diff
@@ -1,6 +1,52 @@
 //SPDX-License-Identifier: Unlicense
-pragma solidity ^0.8.4;
+pragma solidity ^0.8.2;
 
-contract Vault1 {
-    
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "hardhat/console.sol";
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+
+contract DummyToken is ERC20 {
+  constructor(uint256 _amount) ERC20('DummyToken', 'DTN') {
+    _mint(msg.sender, _amount);
+  }
+}
+
+contract Vault1 is ERC20, ReentrancyGuard{
+
+  IERC20 private myToken;
+  mapping(address => uint256) private _balances;
+
+  event Deposited(address _account, uint256 _amount);
+  event Withdrawn(address _account, uint256 _amount);
+
+  constructor(string memory _name, string memory _symbol, IERC20 _token) ERC20(_name, _symbol) {
```

Comment:
This contract need not be ERC20, we should only be taking `_token` address and accessing an already existing ERC20 contract